File: ios/Classes/StreamHandlers/ConnectionStreamHandler.swift
//
//  ConnectionStreamHandler.swift
//  sdk_live_flutter_plugin
//
//  Created by Ross on 29.10.2021.
//

import Foundation
import Flutter
import ScreenMeetSDK

class ConnectionStreamHandler: NSObject, FlutterStreamHandler {
    var sink: FlutterEventSink?
       
    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        sink = events
        return nil
    }
       
     func sendConnectionState(_ connectionState: SMConnectionState) {
        guard let sink = sink else { return }
        sink(connectionState.description)
    }
       
    func onCancel(withArguments arguments: Any?) -> FlutterError? {
        sink = nil
        return nil
    }
}

File: android/src/main/kotlin/com/screenmeet/sdk_live_flutter_plugin/handlers/ConnectionStreamHandler.kt
package com.screenmeet.sdk_live_flutter_plugin.handlers

import com.screenmeet.sdk.ScreenMeet
import io.flutter.plugin.common.EventChannel

class ConnectionStreamHandler: EventChannel.StreamHandler {

    private var sink: EventChannel.EventSink? = null

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) { sink = events }
    override fun onCancel(arguments: Any?) { sink = null }

    fun sendConnectionState(connectionState: ScreenMeet.ConnectionState) {
        sink?.success(connectionState.state.name)
    }
}